pkg/clustermanager/reconcile_logicalcluster.go

```diff
@@ -110,44 +146,35 @@ func (c *controller) processAddLC(newCluster *lcv1alpha1.LogicalCluster) error {
 	return err
 }
 
-/***
-func (c *controller) processUpdateLC(ctx context.Context, key any) error {
-	logger := klog.FromContext(ctx)
-	var err error
-	clusterConfig := key.(*lcv1alpha1.LogicalCluster)
-	_, err = c.clientset.
-		LogicalclusterV1alpha1().
-		LogicalClusters(clusterproviderclient.GetNamespace(clusterConfig.Spec.ClusterProviderDescName)).
-		Update(ctx, clusterConfig, v1.UpdateOptions{})
-	if err != nil {
-		logger.Error(err, "failed to update cluster status.")
+// deleteLC: delete a logical cluster object
+func (c *controller) deleteLC(delCluster *lcv1alpha1.LogicalCluster) error {
+	if delCluster.Spec.Managed {
```

Comment:
But why should you be here at all , in that case?
I suggest just test this condition at the case itself and delete only for managed

Reply:
done